File: app/src/main/java/com/ternaryop/photoshelf/activity/MainActivity.java
package com.ternaryop.photoshelf.activity;

import java.util.List;

import android.app.Fragment;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.widget.AdapterView;
import android.widget.AdapterView.OnItemSelectedListener;
import android.widget.Spinner;
import android.widget.Toast;

import com.ternaryop.photoshelf.AppSupport;
import com.ternaryop.photoshelf.R;
import com.ternaryop.photoshelf.adapter.BlogSpinnerAdapter;
import com.ternaryop.photoshelf.drawer.ItemTestDrawerItem;
import com.ternaryop.photoshelf.event.CounterEvent;
import com.ternaryop.photoshelf.fragment.BestOfFragment;
import com.ternaryop.photoshelf.fragment.BirthdaysBrowserFragment;
import com.ternaryop.photoshelf.fragment.BirthdaysPublisherFragment;
import com.ternaryop.photoshelf.fragment.DraftListFragment;
import com.ternaryop.photoshelf.fragment.FragmentActivityStatus;
import com.ternaryop.photoshelf.fragment.HomeFragment;
import com.ternaryop.photoshelf.fragment.ImagePickerFragment;
import com.ternaryop.photoshelf.fragment.PhotoPreferencesFragment;
import com.ternaryop.photoshelf.fragment.PublishedPostsListFragment;
import com.ternaryop.photoshelf.fragment.SavedContentListFragment;
import com.ternaryop.photoshelf.fragment.ScheduledListFragment;
import com.ternaryop.photoshelf.fragment.TagListFragment;
import com.ternaryop.photoshelf.fragment.TagPhotoBrowserFragment;
import com.ternaryop.photoshelf.service.CounterIntentService;
import com.ternaryop.tumblr.AuthenticationCallback;
import com.ternaryop.tumblr.Tumblr;
import com.ternaryop.utils.DialogUtils;
import com.ternaryop.utils.drawer.activity.DrawerActionBarActivity;
import com.ternaryop.utils.drawer.adapter.DrawerAdapter;
import com.ternaryop.utils.drawer.adapter.DrawerItem;
import io.reactivex.SingleObserver;
import io.reactivex.disposables.Disposable;
import org.greenrobot.eventbus.EventBus;
import org.greenrobot.eventbus.Subscribe;
import org.greenrobot.eventbus.ThreadMode;

public class MainActivity extends DrawerActionBarActivity implements AuthenticationCallback, FragmentActivityStatus {
    private static final String LOADER_PREFIX_AVATAR = "avatar";

    private final static int DRAWER_ITEM_HOME = 0;
    private final static int DRAWER_ITEM_DRAFT = 1;
    private final static int DRAWER_ITEM_SCHEDULE = 2;
    private final static int DRAWER_ITEM_PUBLISHED_POST = 3;
    private final static int DRAWER_ITEM_BROWSE_IMAGES_BY_TAGS = 4;
    private final static int DRAWER_ITEM_BROWSE_TAGS = 5;
    private final static int DRAWER_ITEM_BIRTHDAYS_BROWSER = 6;
    private final static int DRAWER_ITEM_BIRTHDAYS_TODAY = 7;
    private final static int DRAWER_ITEM_BEST_OF = 8;
    private final static int DRAWER_ITEM_TEST_PAGE = 9;
    private final static int DRAWER_ITEM_SETTINGS = 10;
    private final static int DRAWER_ITEM_FEEDLY = 11;

    private AppSupport appSupport;
    private Spinner blogList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        appSupport = new AppSupport(this);
        rebuildDrawerMenu();

        blogList = (Spinner) findViewById(R.id.blogs_spinner);
        blogList.setOnItemSelectedListener(new BlogItemSelectedListener());

        boolean logged = Tumblr.isLogged(this);
        if (savedInstanceState == null) {
            if (logged) {
                showHome();
            } else {
                showSettings();
            }
        }
        enableUI(logged);
    }

    @Override
    public void onStart() {
        super.onStart();
        EventBus.getDefault().register(this);
    }

    @Override
    public void onStop() {
        EventBus.getDefault().unregister(this);
        super.onStop();
    }

    @Override
    protected int getActivityLayoutResId() {
        return R.layout.activity_main;
    }

    private void showHome() {
        selectClickedItem(0);
        openDrawer();
    }

    private void showSettings() {
        selectClickedItem(getAdapter().getCount() - 1);
    }

    @Override
    protected DrawerAdapter initDrawerAdapter() {
        DrawerAdapter adapter = new DrawerAdapter(this);

        adapter.add(new DrawerItem(DRAWER_ITEM_HOME, getString(R.string.home), HomeFragment.class));

        adapter.add(new DrawerItem(DRAWER_ITEM_DRAFT, getString(R.string.draft_title), DraftListFragment.class, true));
        adapter.add(new DrawerItem(DRAWER_ITEM_SCHEDULE, getString(R.string.schedule_title), ScheduledListFragment.class, true));
        adapter.add(new DrawerItem(DRAWER_ITEM_PUBLISHED_POST, getString(R.string.published_post), PublishedPostsListFragment.class));

        // Tags
        adapter.add(DrawerItem.DRAWER_ITEM_DIVIDER);
        adapter.add(new DrawerItem(DRAWER_ITEM_BROWSE_IMAGES_BY_TAGS, getString(R.string.browse_images_by_tags_title), TagPhotoBrowserFragment.class));
        adapter.add(new DrawerItem(DRAWER_ITEM_BROWSE_TAGS, getString(R.string.browse_tags_title), TagListFragment.class));

        // Extras
        adapter.add(DrawerItem.DRAWER_ITEM_DIVIDER);
        adapter.add(new DrawerItem(DRAWER_ITEM_BIRTHDAYS_BROWSER, getString(R.string.birthdays_browser_title), BirthdaysBrowserFragment.class));
        adapter.add(new DrawerItem(DRAWER_ITEM_BIRTHDAYS_TODAY, getString(R.string.birthdays_today_title), BirthdaysPublisherFragment.class,  true));
        adapter.add(new DrawerItem(DRAWER_ITEM_BEST_OF, getString(R.string.best_of), BestOfFragment.class));
        adapter.add(new DrawerItem(DRAWER_ITEM_FEEDLY, "Feedly", SavedContentListFragment.class));
        adapter.add(new ItemTestDrawerItem(DRAWER_ITEM_TEST_PAGE, getString(R.string.test_page_title), ImagePickerFragment.class));

        // Settings
        adapter.add(DrawerItem.DRAWER_ITEM_DIVIDER);
        adapter.add(new DrawerItem(DRAWER_ITEM_SETTINGS, getString(R.string.settings), PhotoPreferencesFragment.class));

        return adapter;
    }

    @Override
    protected void onResume() {
        super.onResume();

        if (!Tumblr.isLogged(this)) {
            // if we are returning from authentication then enable the UI
            boolean handled = Tumblr.handleOpenURI(this, getIntent().getData(), this);

            // show the preference only if we aren't in the middle of URI handling and not already logged in
            if (handled) {
                showHome();
            } else {
                showSettings();
            }
        }
    }

    private class BlogItemSelectedListener implements OnItemSelectedListener {
        @Override
        public void onItemSelected(AdapterView<?> parent, View view, int pos, long id) {
            String blogName = (String) blogList.getSelectedItem();
            appSupport.setSelectedBlogName(blogName);
            refreshCounters(blogName);
        }

        public void refreshCounters(String blogName) {
            CounterIntentService.fetchCounter(MainActivity.this, blogName, CounterEvent.BIRTHDAY);
            CounterIntentService.fetchCounter(MainActivity.this, blogName, CounterEvent.DRAFT);
            CounterIntentService.fetchCounter(MainActivity.this, blogName, CounterEvent.SCHEDULE);
        }

        public void onNothingSelected(AdapterView<?> parent) {
        }
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    public void onCounterEvent(CounterEvent event) {
        String value = event.getCount() > 0 ? String.valueOf(event.getCount()) : null;

        switch (event.getType()) {
            case CounterEvent.BIRTHDAY:
                getAdapter().getItemById(DRAWER_ITEM_BIRTHDAYS_TODAY).setBadge(value);
                break;
            case CounterEvent.DRAFT:
                getAdapter().getItemById(DRAWER_ITEM_DRAFT).setBadge(value);
                break;
            case CounterEvent.SCHEDULE:
                getAdapter().getItemById(DRAWER_ITEM_SCHEDULE).setBadge(value);
                break;
            case CounterEvent.NONE:
                break;
        }
        getAdapter().notifyDataSetChanged();
    }

    private void enableUI(boolean enabled) {
        if (enabled) {
            appSupport.fetchBlogNames(this)
                    .subscribe(new SingleObserver<List<String>>() {
                        @Override
                        public void onSubscribe(Disposable d) {

                        }

                        @Override
                        public void onSuccess(List<String> blogNames) {
                            fillBlogList(blogNames);
                        }

                        @Override
                        public void onError(Throwable e) {
                            DialogUtils.showErrorDialog(getApplicationContext(), e);
                        }
                    });
        }
        getDrawerToggle().setDrawerIndicatorEnabled(enabled);
        getAdapter().setSelectionEnabled(enabled);
        getAdapter().notifyDataSetChanged();
    }

    @Override
    public void tumblrAuthenticated(final String token, final String tokenSecret, final Exception error) {
        if (error == null) {
            Toast.makeText(this,
                    getString(R.string.authentication_success_title),
                    Toast.LENGTH_LONG)
                    .show();
            // after authentication cache blog names
            appSupport.fetchBlogNames(this)
                    .subscribe(new SingleObserver<List<String>>() {
                        @Override
                        public void onSubscribe(Disposable d) {

                        }

                        @Override
                        public void onSuccess(List<String> value) {
                            enableUI(token != null && tokenSecret != null);
                        }

                        @Override
                        public void onError(Throwable e) {
                            DialogUtils.showErrorDialog(MainActivity.this, e);
                        }
                    });
        } else {
            DialogUtils.showErrorDialog(this, error);
        }
    }

    public String getBlogName() {
        return appSupport.getSelectedBlogName();
    }

    @Override
    protected void onDrawerItemSelected(DrawerItem drawerItem) {
        try {
            Fragment fragment = drawerItem.instantiateFragment(getApplicationContext());
            getFragmentManager().beginTransaction().replace(R.id.content_frame, fragment).commit();
        } catch (Exception e) {
            DialogUtils.showErrorDialog(getApplication(), e);
            e.printStackTrace();
        }
    }

    private void fillBlogList(List<String> blogNames) {
        BlogSpinnerAdapter adapter = new BlogSpinnerAdapter(this, LOADER_PREFIX_AVATAR, blogNames);
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        blogList.setAdapter(adapter);

        String selectedName = appSupport.getSelectedBlogName();
        if (selectedName != null) {
            int position = adapter.getPosition(selectedName);
            if (position >= 0) {
                blogList.setSelection(position);
            }
        }
    }

    @Override
    public AppSupport getAppSupport() {
        return appSupport;
    }

    @Override
    public Toolbar getToolbar() {
        return (Toolbar) findViewById(R.id.drawer_toolbar);
    }
}

File: app/src/main/java/com/ternaryop/photoshelf/activity/MainActivity.kt
@file:Suppress("MaxLineLength")

package com.ternaryop.photoshelf.activity

import android.content.SharedPreferences
import android.os.Bundle
import android.view.MenuItem
import android.view.View
import android.widget.AdapterView
import android.widget.AdapterView.OnItemSelectedListener
import android.widget.Spinner
import android.widget.Toast
import androidx.activity.viewModels
import androidx.appcompat.widget.Toolbar
import androidx.core.os.bundleOf
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import androidx.preference.PreferenceManager
import androidx.preference.PreferenceScreen
import com.ternaryop.photoshelf.R
import com.ternaryop.photoshelf.birthday.browser.fragment.BirthdayBrowserFragment
import com.ternaryop.photoshelf.birthday.publisher.fragment.BirthdayPublisherFragment
import com.ternaryop.photoshelf.core.prefs.selectedBlogName
import com.ternaryop.photoshelf.feedly.fragment.FeedlyContentType
import com.ternaryop.photoshelf.feedly.fragment.FeedlyListFragment
import com.ternaryop.photoshelf.fragment.BestOfFragment
import com.ternaryop.photoshelf.fragment.FragmentActivityStatus
import com.ternaryop.photoshelf.fragment.appFragmentFactory
import com.ternaryop.photoshelf.fragment.preference.MainPreferenceFragment
import com.ternaryop.photoshelf.fragment.preference.PreferenceCategorySelector
import com.ternaryop.photoshelf.home.fragment.HomeFragment
import com.ternaryop.photoshelf.imagepicker.fragment.EXTRA_URL
import com.ternaryop.photoshelf.imagepicker.fragment.ImagePickerFragment
import com.ternaryop.photoshelf.lifecycle.EventObserver
import com.ternaryop.photoshelf.lifecycle.Status
import com.ternaryop.photoshelf.tagnavigator.fragment.TagListFragment
import com.ternaryop.photoshelf.tagphotobrowser.fragment.TagPhotoBrowserFragment
import com.ternaryop.photoshelf.tumblr.dialog.blog.BlogSpinnerAdapter
import com.ternaryop.photoshelf.tumblr.ui.draft.fragment.DraftListFragment
import com.ternaryop.photoshelf.tumblr.ui.publish.fragment.PublishedPostsListFragment
import com.ternaryop.photoshelf.tumblr.ui.schedule.fragment.ScheduledListFragment
import com.ternaryop.photoshelf.util.askNotificationsPermission
import com.ternaryop.tumblr.android.TumblrManager
import com.ternaryop.utils.dialog.showErrorDialog
import com.ternaryop.utils.drawer.activity.DrawerActionBarActivity
import com.ternaryop.utils.drawer.adapter.DrawerAdapter
import com.ternaryop.utils.drawer.adapter.DrawerItem
import dagger.hilt.android.AndroidEntryPoint

private const val ASK_NOTIFICATION_PERMISSION = 2

@AndroidEntryPoint
class MainActivity :
    DrawerActionBarActivity(),
    FragmentActivityStatus,
    PreferenceFragmentCompat.OnPreferenceStartScreenCallback,
    PreferenceFragmentCompat.OnPreferenceStartFragmentCallback {
    private lateinit var blogList: Spinner
    private lateinit var prefs: SharedPreferences

    private val viewModel: MainActivityViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        supportFragmentManager.fragmentFactory = appFragmentFactory
        super.onCreate(savedInstanceState)

        prefs = PreferenceManager.getDefaultSharedPreferences(this)

        rebuildDrawerMenu()

        blogList = findViewById(R.id.blogs_spinner)
        blogList.onItemSelectedListener = BlogItemSelectedListener()

        viewModel.result.observe(
            this,
            EventObserver { result ->
                when (result) {
                    is MainActivityModelResult.BirthdaysCount ->
                        onUpdateCount(DRAWER_ITEM_BIRTHDAYS_TODAY, result.command.data ?: 0)
                    is MainActivityModelResult.DraftCount ->
                        onUpdateCount(DRAWER_ITEM_DRAFT, result.command.data ?: 0)
                    is MainActivityModelResult.ScheduledCount ->
                        onUpdateCount(DRAWER_ITEM_SCHEDULE, result.command.data ?: 0)
                    is MainActivityModelResult.BlogNames -> onBlogNames(result)
                    is MainActivityModelResult.TumblrAuthenticated -> onTumblrAuthenticated(result)
                }
            }
        )

        val logged = TumblrManager.isLogged(this)
        if (savedInstanceState == null) {
            onAppStarted(logged)
        }
        enableUI(logged)

        askNotificationsPermission(this, R.string.notification_permission_rationale, ASK_NOTIFICATION_PERMISSION)
    }

    private fun onAppStarted(logged: Boolean) {
        if (logged) {
            if (!handleShortcutAction()) {
                showHome()
            }
        } else {
            showSettings()
        }
    }

    override val activityLayoutResId: Int
        get() = R.layout.activity_main

    private fun showHome() {
        selectClickedItem(0)
        openDrawer()
    }

    private fun handleShortcutAction(): Boolean {
        when (intent.action) {
            SHORTCUT_ACTION_DRAFT -> selectClickedItem(DRAWER_ITEM_DRAFT)
            SHORTCUT_ACTION_SCHEDULED -> selectClickedItem(DRAWER_ITEM_SCHEDULE)
            SHORTCUT_ACTION_FEEDLY_SAVED -> selectClickedItem(DRAWER_ITEM_FEEDLY_READ_LATER)
            SHORTCUT_ACTION_FEEDLY_UNREAD -> selectClickedItem(DRAWER_ITEM_FEEDLY_UNREAD)
            else -> return false
        }
        return true
    }

    private fun showSettings() {
        selectClickedItem(adapter.count - 1)
    }

    override fun initDrawerAdapter(): DrawerAdapter {
        val adapter = DrawerAdapter(this)

        adapter.add(DrawerItem(DRAWER_ITEM_HOME, getString(R.string.home), HomeFragment::class.java))

        adapter.add(DrawerItem(DRAWER_ITEM_DRAFT, getString(com.ternaryop.photoshelf.tumblr.dialog.R.string.draft_title), DraftListFragment::class.java, true))
        adapter.add(DrawerItem(DRAWER_ITEM_SCHEDULE, getString(R.string.schedule_title), ScheduledListFragment::class.java, true))
        adapter.add(DrawerItem(DRAWER_ITEM_PUBLISHED_POST, getString(R.string.published_post), PublishedPostsListFragment::class.java))
        // Tags
        adapter.add(DrawerItem.DRAWER_ITEM_DIVIDER)
        adapter.add(DrawerItem(DRAWER_ITEM_BROWSE_IMAGES_BY_TAGS, getString(R.string.browse_images_by_tags_title), TagPhotoBrowserFragment::class.java))

        adapter.add(
            DrawerItem(
                DRAWER_ITEM_BROWSE_TAGS, getString(R.string.browse_tags_title), TagListFragment::class.java,
                argumentsBuilder = {
                    prefs.selectedBlogName?.let { bundleOf(TagListFragment.ARG_BLOG_NAME to it) }
                }
            )
        )

        // Extras
        adapter.add(DrawerItem.DRAWER_ITEM_DIVIDER)
        adapter.add(DrawerItem(DRAWER_ITEM_BIRTHDAYS_BROWSER, getString(R.string.birthdays_browser_title), BirthdayBrowserFragment::class.java))
        adapter.add(DrawerItem(DRAWER_ITEM_BIRTHDAYS_TODAY, getString(R.string.birthdays_today_title), BirthdayPublisherFragment::class.java, true))
        adapter.add(DrawerItem(DRAWER_ITEM_BEST_OF, getString(R.string.best_of), BestOfFragment::class.java))
        adapter.add(DrawerItem(DRAWER_ITEM_FEEDLY_READ_LATER, getString(R.string.feedly_saved_title), FeedlyListFragment::class.java))
        adapter.add(
            DrawerItem(
                DRAWER_ITEM_FEEDLY_UNREAD,
                getString(R.string.feedly_unread_title), FeedlyListFragment::class.java,
                arguments = bundleOf(FeedlyListFragment.ARG_CONTENT_TYPE to FeedlyContentType.Unread)
            )
        )

        adapter.add(
            DrawerItem(
                DRAWER_ITEM_TEST_PAGE,
                getString(R.string.test_page_title), ImagePickerFragment::class.java, arguments = bundleOf(EXTRA_URL to getString(R.string.test_page_url))
            )
        )
        // Settings
        adapter.add(DrawerItem.DRAWER_ITEM_DIVIDER)
        adapter.add(DrawerItem(DRAWER_ITEM_SETTINGS, getString(com.ternaryop.photoshelf.feedly.R.string.settings), MainPreferenceFragment::class.java))

        return adapter
    }

    override fun onResume() {
        super.onResume()

        viewModel.handleCallbackTumblrUri(intent.data)
    }

    private inner class BlogItemSelectedListener : OnItemSelectedListener {
        override fun onItemSelected(parent: AdapterView<*>?, view: View?, pos: Int, id: Long) {
            val blogName = blogList.selectedItem as String
            if (blogName != prefs.selectedBlogName) {
                prefs.selectedBlogName = blogName
                viewModel.clearCounters()
            }
            viewModel.birthdaysCount()
            viewModel.draftCount(blogName)
            viewModel.scheduledCount(blogName)
        }

        override fun onNothingSelected(parent: AdapterView<*>?) = Unit
    }

    private fun onUpdateCount(itemId: Int, count: Int) {
        adapter.getItemById(itemId)?.badge = if (count > 0) count.toString() else null
        adapter.notifyDataSetChanged()
    }

    private fun onBlogNames(result: MainActivityModelResult.BlogNames) {
        when (result.command.status) {
            Status.SUCCESS -> result.command.data?.also { blogNames -> fillBlogList(blogNames) }
            Status.ERROR -> result.command.error?.also { it.showErrorDialog(this) }
            Status.PROGRESS -> {}
        }
    }

    private fun onTumblrAuthenticated(result: MainActivityModelResult.TumblrAuthenticated) {
        when (result.command.status) {
            Status.SUCCESS -> result.command.data?.also { handled ->
                if (handled) {
                    tumblrAuthenticated()
                    showHome()
                } else {
                    showSettings()
                }
            }
            Status.ERROR -> result.command.error?.also { it.showErrorDialog(this) }
            Status.PROGRESS -> {}
        }
    }

    private fun enableUI(enabled: Boolean) {
        if (enabled) {
            viewModel.fetchBlogNames()
        }
        drawerToggle.isDrawerIndicatorEnabled = enabled
        adapter.isSelectionEnabled = enabled
        adapter.notifyDataSetChanged()
    }

    private fun tumblrAuthenticated() {
        Toast.makeText(
            this,
            getString(R.string.authentication_success_title),
            Toast.LENGTH_LONG
        ).show()
        enableUI(true)
    }

    override fun onDrawerItemSelected(drawerItem: DrawerItem) {
        try {
            val fragment = drawerItem.instantiateFragment(applicationContext, supportFragmentManager)
            supportFragmentManager.beginTransaction().replace(R.id.content_frame, fragment).commit()
        } catch (e: Exception) {
            e.showErrorDialog(application)
            e.printStackTrace()
        }
    }

    private fun fillBlogList(blogNames: List<String>) {
        val adapter = BlogSpinnerAdapter(this, blogNames)
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        blogList.adapter = adapter

        val selectedName = prefs.selectedBlogName
        if (selectedName != null) {
            val position = adapter.getPosition(selectedName)
            if (position >= 0) {
                blogList.setSelection(position)
            }
        }
    }

    override val isDrawerMenuOpen: Boolean
        get() = this.isDrawerOpen

    override val drawerToolbar: Toolbar
        get() = toolbar

    override val toolbar: Toolbar
        get() = findViewById<View>(com.ternaryop.photoshelf.imageviewer.R.id.drawer_toolbar) as Toolbar

    override fun onPreferenceStartScreen(
        caller: PreferenceFragmentCompat,
        pref: PreferenceScreen
    ): Boolean {
        val fragment = PreferenceCategorySelector.fragmentFromCategory(pref.key) ?: return false
        return PreferenceCategorySelector.openScreen(caller, pref, fragment)
    }

    override fun onPreferenceStartFragment(
        caller: PreferenceFragmentCompat,
        pref: Preference
    ): Boolean {
        val prefFragment = pref.fragment ?: return false
        val fragment = supportFragmentManager.fragmentFactory.instantiate(
            classLoader,
            prefFragment
        )
        return PreferenceCategorySelector.openScreen(caller, pref, fragment)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // when PreferenceScreen opens a new fragment isn't possible to to come back
        // using the hamburger menu, so we check manually if we are on some settings fragment
        if (supportFragmentManager.fragments.size > 1 && supportFragmentManager.fragments[0] is MainPreferenceFragment) {
            supportFragmentManager.popBackStack()
            return true
        }

        return super.onOptionsItemSelected(item)
    }

    companion object {
        private const val DRAWER_ITEM_HOME = 0
        private const val DRAWER_ITEM_DRAFT = 1
        private const val DRAWER_ITEM_SCHEDULE = 2
        private const val DRAWER_ITEM_PUBLISHED_POST = 3
        private const val DRAWER_ITEM_BROWSE_IMAGES_BY_TAGS = 4
        private const val DRAWER_ITEM_BROWSE_TAGS = 5
        private const val DRAWER_ITEM_BIRTHDAYS_BROWSER = 6
        private const val DRAWER_ITEM_BIRTHDAYS_TODAY = 7
        private const val DRAWER_ITEM_BEST_OF = 8
        private const val DRAWER_ITEM_TEST_PAGE = 9
        private const val DRAWER_ITEM_SETTINGS = 10
        private const val DRAWER_ITEM_FEEDLY_READ_LATER = 11
        private const val DRAWER_ITEM_FEEDLY_UNREAD = 12

        const val SHORTCUT_ACTION_DRAFT = "com.ternaryop.photoshelf.shortcut.draft"
        const val SHORTCUT_ACTION_SCHEDULED = "com.ternaryop.photoshelf.shortcut.scheduled"
        const val SHORTCUT_ACTION_FEEDLY_SAVED = "com.ternaryop.photoshelf.shortcut.feedly.saved"
        const val SHORTCUT_ACTION_FEEDLY_UNREAD = "com.ternaryop.photoshelf.shortcut.feedly.unread"
    }
}
